riptide-failsafe/src/main/java/org/zalando/riptide/failsafe/FailsafePlugin.java

```diff
@@ -30,33 +28,45 @@
 
     public static final Attribute<Integer> ATTEMPTS = Attribute.generate();
 
-    private final ImmutableList<? extends Policy<ClientHttpResponse>> policies;
+    private final ImList<RequestPolicy> policies;
     private final TaskDecorator decorator;
-    private final Predicate<RequestArguments> predicate;
-    private final RetryListener listener;
 
-    public FailsafePlugin(final ImmutableList<? extends Policy<ClientHttpResponse>> policies) {
-        this(policies, identity(), new IdempotencyPredicate(), RetryListener.DEFAULT);
+    public FailsafePlugin() {
+        this(vec(), identity());
     }
 
-    public FailsafePlugin withDecorator(final TaskDecorator decorator) {
-        return new FailsafePlugin(policies, decorator, predicate, listener);
+    public FailsafePlugin withPolicy(final BackupRequest<ClientHttpResponse> policy) {
+        return withPolicy(policy, new IdempotencyPredicate());
+    }
+
+    public FailsafePlugin withPolicy(final RetryPolicy<ClientHttpResponse> policy) {
+        return withPolicy(new RetryRequestPolicy(policy));
+    }
```

Comment:
Those two are only here to prevent users from shooting themselves in the foot and forgetting to specify a conditional predicate. Ideally there would be a cleaner way to do that.